tools/SwiftSyntax/CMakeLists.txt

```diff
@@ -1,5 +1,29 @@
+set(swiftsyntax_compile_flags "${SWIFT_RUNTIME_SWIFT_COMPILE_FLAGS}")
+
+# FIXME: This is _awful_. These pieces could/should be sunk into
+#        `add_swift_library`
+if("${SWIFT_HOST_VARIANT_SDK}" STREQUAL "LINUX")
+  LIST(APPEND swiftsyntax_compile_flags
+    -I"${SWIFT_PATH_TO_LIBDISPATCH_SOURCE}"
+    -I"${SWIFT_PATH_TO_LIBDISPATCH_BUILD}/src"
+    -I"${SWIFT_PATH_TO_LIBDISPATCH_BUILD}/src/swift"
+    -I"${FOUNDATION_BUILD_DIR}"
+    -I"${FOUNDATION_BUILD_DIR}/usr/lib/swift"
+    -I"${FOUNDATION_BUILD_DIR}/usr/lib/swift/CoreFoundation"
+
+    -L"${SWIFT_PATH_TO_LIBDISPATCH_BUILD}/src"
+    -L"${SWIFT_PATH_TO_LIBDISPATCH_BUILD}/src/swift"
+    -L"${FOUNDATION_BUILD_DIR}"
+    -L"${FOUNDATION_BUILD_DIR}/usr/lib/swift"
+    -L"${FOUNDATION_BUILD_DIR}/usr/lib/swift/CoreFoundation"
```

Comment:
Please use the `swift_target_link_search_directories` function for these search paths.